packages/fast-components-styles-msft/src/utilities/color/accessible-recipe.ts

```diff
@@ -0,0 +1,95 @@
+import {
+    checkDesignSystemResolver,
+    DesignSystem,
+    DesignSystemResolver,
+} from "../../design-system";
+import { backgroundColor } from "../design-system";
+import { Swatch, SwatchFamily } from "./common";
+import {
+    findSwatchIndex,
+    getSwatch,
+    isDarkMode,
+    minContrastTargetFactory,
+    Palette,
+    swatchByContrast,
+} from "./palette";
+
+function indexToSwatchFamily(
+    accessibleIndex: number,
+    palette: Palette,
+    direction: number,
+    restDelta: number,
+    hoverDelta: number,
+    activeDelta: number
+): SwatchFamily {
+    // One of the indexes will be rest, the other will be hover. Depends on the offsets and the direction.
+    const accessibleIndex2: number =
+        accessibleIndex + direction * Math.abs(restDelta - hoverDelta);
+
+    const indexOneIsRestState: boolean =
+        direction === 1
+            ? restDelta < hoverDelta
+            : direction * restDelta > direction * hoverDelta;
+
+    const restIndex: number = indexOneIsRestState ? accessibleIndex : accessibleIndex2;
+    const hoverIndex: number = indexOneIsRestState ? accessibleIndex2 : accessibleIndex;
+
+    const activeIndex: number = restIndex + direction * activeDelta;
+
+    return {
+        rest: getSwatch(restIndex, palette),
+        hover: getSwatch(hoverIndex, palette),
+        active: getSwatch(activeIndex, palette),
+    };
+}
+
+/**
+ * Function to derive accessible colors from contrast and delta configuration.
+ * Performs a simple contrast check against the colors and returns
+ * the color that has the most contrast against the background. If contrast
+ * cannot be retrieved correctly, function returns black.
+ */
+export function accessibleAlgorithm(
+    palette: Palette | DesignSystemResolver<Palette>,
+    minContrast: number | DesignSystemResolver<number>,
+    restDelta: number | DesignSystemResolver<number>,
+    hoverDelta: number | DesignSystemResolver<number>,
+    activeDelta: number | DesignSystemResolver<number>
+): DesignSystemResolver<SwatchFamily> {
+    return (designSystem: DesignSystem): SwatchFamily => {
+        const resolvedPalette: Palette = checkDesignSystemResolver(palette, designSystem);
+        const direction: 1 | -1 = isDarkMode(designSystem) ? -1 : 1;
+
+        const accessibleSwatch: Swatch = swatchByContrast(
```

Comment:
This should be defined outside of the returned callback to prevent a new function from being created with the same arguments every invocation.

Reply:
Per our chat, this should only be called once due to memoization.